src/main/java/com/spring/cinema/dao/impl/AbstractDao.java

```diff
@@ -1,21 +1,26 @@
-package com.dmgrt.cinema.dao.impl;
+package com.spring.cinema.dao.impl;
 
-import com.dmgrt.cinema.exceptions.DataProcessingException;
-import com.dmgrt.cinema.util.HibernateUtil;
+import com.spring.cinema.exceptions.DataProcessingException;
 import java.util.List;
 import org.apache.log4j.Logger;
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
 import org.hibernate.query.Query;
 
 public class AbstractDao<T> {
     private static final Logger logger = Logger.getLogger(AbstractDao.class);
+    private final SessionFactory sessionFactory;
+
```

Comment:
maybe you skipped @Autowired here?

Reply:
If I remember clearly, we have no need to set annotations when we use the constructor